specs.js

```diff
@@ -112,75 +135,571 @@ describe('when using trimperson api', () => {
+      })
+
+      When((done) => {
+        trim.getDocument(documentId, function (err, data) {
           dataReturn = data
+          errReturn = err
           done()
         })
       })
-      Then('it should be able to succesfully use the function', () => {
+
+      Then('It should return an error', (done) => {
+        expect(errReturn).to.exist
+        expect(dataReturn).not.to.exist
+        getDocumentMock.done()
+        done()
+      })
+    })
+  })
+
+  /*
+   CREATE CONTAINER
+   */
+  describe('To create a container', () => {
+    Then('`createContainer` should be a function', (done) => {
+      expect(trim.createContainer).to.be.a('function')
+      done()
+    })
+    Then('`createPublicContainer` should be a function', (done) => {
+      expect(trim.createPublicContainer).to.be.a('function')
+      done()
+    })
+    Then('`createPrivateContainer` should be a function', (done) => {
+      expect(trim.createPrivateContainer).to.be.a('function')
+      done()
+    })
+
+    describe('If all inputs are valid', () => {
+      var createContainerMock, dataReturn, errReturn, data
+
+      Given(() => data = {
+        folderName: VALID_FOLDER_NAME,
+        privacyLevel: trim.PRIVACY_LEVELS.PUBLIC
+      })
+      Given(() => {
+        createContainerMock = nock(mockURL)
+            .filteringRequestBody(R.always('*'))
+            .post('/CreateContainer', '*')
+            .query({securityToken: mockToken})
+            .reply(201, {
+              RecordNo: VALID_FOLDER_NAME
+            })
+      })
+      When(done => {
+        trim.createContainer(data, (err, body) => {
+          errReturn = err
+          dataReturn = body
+          done()
+        })
+      })
+      Then('The container is created', done => {
+        expect(errReturn).not.to.exist
+        expect(dataReturn).to.be.a('object')
+        expect(dataReturn.RecordNo).to.equal(VALID_FOLDER_NAME)
+        createContainerMock.done()
+        done()
+      })
+    })
+
+    describe('If all no privacy is specified', () => {
+      var createContainerMock, dataReturn, errReturn, data
+
+      Given(() => data = {
+        folderName: VALID_FOLDER_NAME
+      })
+      Given(() => {
+        createContainerMock = nock(mockURL)
+            .filteringRequestBody(R.always('*'))
+            .post('/CreateContainer', '*')
+            .query({securityToken: mockToken})
+            .reply(201, {
+              RecordNo: VALID_FOLDER_NAME
+            })
+      })
+      When(done => {
+        trim.createContainer(data, (err, body) => {
+          errReturn = err
+          dataReturn = body
+          done()
+        })
+      })
+      Then('The container is created', done => {
+        expect(errReturn).not.to.exist
+        expect(dataReturn).to.be.a('object')
+        expect(dataReturn.RecordNo).to.equal(VALID_FOLDER_NAME)
+        createContainerMock.done()
+        done()
+      })
+    })
+
+    describe('No folder name is provided', () => {
```

Comment:
I just confirmed - you can have containers with a colon in the name. The only invalid title is empty string AFIK. I bet there are other invalid names, but I can't find any. 

Also, creating a container with name `"Bad Name!@#$%^&*()"` returns
```json
{
  "RecordNo": "BAD NAME!@"
}
```

But I just realized that the api strips out the # and anything after. I think it's because subfolders have the form e.g. `PN#03`, so it's looking for a subfolder. I can't actually create a subfolder though...

@digitalsadhu What are your thoughts on validating this? It's not good that the api is stripping out characters from the folder names...